### PR TITLE
Feature/time logs

### PR DESCRIPTION
**Pull request by sidra-kidwai**
Added time logs and views updated for projects.
Also some minor code fixes.

### CODE REVIEW

File: app/controllers/projects_controller.rb [hunk 1]
```diff
@@ -10,7 +10,7 @@ def index
 
   def show
     @comments = @project.comments
-    @payments = @project.payments
+    @time_logs = @project.time_logs
```
Comment (AmmadJavaid) [inline app/controllers/projects_controller.rb]: I think we don't need the instance variable here. Just use  `@project.time_logs` where needed.

File: app/controllers/time_logs_controller.rb [hunk 2]
```diff
@@ -0,0 +1,46 @@
+# frozen_string_literal: true
+
+class TimeLogsController < ApplicationController
+  before_action :set_project
+  before_action :set_time_log, only: %i[edit update destroy]
+
+  def index
+    @time_logs = @project.time_logs
```
Comment (AmmadJavaid) [inline app/controllers/time_logs_controller.rb]: why `@time_logs` ?

File: app/controllers/time_logs_controller.rb [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+
+  def new
+    @time_logs = @project.time_logs
+    @time_log = @project.time_logs.new
+  end
+
+  def create
+    @time_log = @project.time_logs.new(time_log_params)
+    @time_log.user_id = current_user.id
+    flash.now[:success] = 'Time-log was successfully added!' if @time_log.save
```
Comment (AmmadJavaid) [inline app/controllers/time_logs_controller.rb:22]: I think we are missing error message here?

File: app/controllers/time_logs_controller.rb [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+  end
+
+  def edit; end
+
+  def update
+    @time_log.update(time_log_params)
```
Comment (AmmadJavaid) [inline app/controllers/time_logs_controller.rb:28]: missing user alert?

File: app/models/time_log.rb [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+  belongs_to :user
+  belongs_to :project
+
+  validates :starting_time, presence: true
+  validates :ending_time, presence: true
+  validate :check_start_time,
+           :check_ending_time
+
+  def check_start_time
+    errors.add(:starting_time, "can't be greater than ending time.") if starting_time.present? && starting_time > ending_time
```
Comment (AmmadJavaid) [inline app/models/time_log.rb]: what about if ending_time is `nil` ?

File: app/controllers/time_logs_controller.rb (hunk 2 above)
Comment (AmmadJavaid) [reply app/controllers/time_logs_controller.rb]: include the `user` here. 
https://github.com/sidra-kidwai/project-management-app/pull/3/files#diff-36a787df6aec4349d345ae3b297a5eccR2

File: app/views/time_logs/_time_log.html.erb [hunk 6]
```diff
@@ -0,0 +1,8 @@
+<tr>
+  <td>Hours logged by <%= time_log.user.name %> on <%= date_format('date', time_log.created_at) %> at <%= date_format('time', time_log.created_at) %></td>
+  <td class='time-log-<%= time_log.id %>'><%= time_log.hours_spent %></td>
+  <td>
+    <%= link_to fa_icon('pencil', text: 'Edit'), edit_project_time_log_path(project, time_log), remote: true, class: "btn btn-primary btn-xs" if policy(time_log).edit? %>
+    <%= link_to fa_icon('trash', text: 'Delete'), [project, time_log], method: :delete, data: { confirm: "Are you sure?" }, class: "btn btn-danger btn-xs" if policy(time_log).destroy? %>
```
Comment (AmmadJavaid) [inline app/views/time_logs/_time_log.html.erb:6]: as we rendering an icon here. Why we need these classes `btn btn-danger`

File: app/views/time_logs/edit.js.erb [hunk 7]
```diff
@@ -0,0 +1,19 @@
+$("#modal-display").html("<%= j render "form" %>");
+$(function () {
```
Comment (AmmadJavaid) [inline app/views/time_logs/edit.js.erb]: This datepicker JS part is same in `edit.js.erb` and `new.js.erb`. Please move this to a separate file and call here.

File: app/views/time_logs/new.js.erb [hunk 8]
```diff
@@ -0,0 +1,18 @@
+$("#modal-display").html("<%= j render "form" %>");
```
Comment (AmmadJavaid) [inline app/views/time_logs/new.js.erb:1]: let discuss this verbally.

**Review by AmmadJavaid — CHANGES_REQUESTED** [review]

File: app/views/time_logs/edit.js.erb (hunk 7 above)
Comment (sidra-kidwai) [reply app/views/time_logs/edit.js.erb]: Alright.

**Pull request merged**